pyannotate_tools/fixes/tests/test_annotate_json_py2.py

```diff
@@ -60,6 +60,102 @@ def nop(foo, bar):
             """
         self.check(a, b)
 
+    def test_decorator_func(self):
+        self.setTestData(
+            [{"func_name": "foo",
+              "path": "<string>",
+              "line": 2,
+              "signature": {
+                  "arg_types": [],
+                  "return_type": "int"},
+              }])
+        a = """\
+            @dec
+            def foo():
+                return 42
+            """
+        b = """\
+            @dec
+            def foo():
+                # type: () -> int
```

Comment:
What happens if the decorator does not preserve the signature?

What happens to argument types?

Reply:
This is the pyannotate fixer. Its job is to just insert type annotations that are handed to it. That's a question for whatever the frontend producing the annotation is.